net-p2p/bitcoin-qt/bitcoin-qt-0.16.0.ebuild

```diff
@@ -0,0 +1,232 @@
+	else
+		einfo "Replace By Fee policy is disabled: Your node will only accept the first transaction seen consuming a conflicting input, regardless of fee offered by later ones."
+	fi
+}
+
+src_prepare() {
+	sed -i 's/^\(complete -F _bitcoind \)bitcoind \(bitcoin-qt\)$/\1\2/' contrib/bitcoind.bash-completion || die
+
+	local KNOTS_PATCHDIR="${WORKDIR}/${KNOTS_P}.patches/"
+
+	eapply "${KNOTS_PATCHDIR}/${KNOTS_P}.syslibs.patch"
+
+	if use knots; then
+		eapply "${KNOTS_PATCHDIR}/${KNOTS_P}.f.patch"
+		eapply "${KNOTS_PATCHDIR}/${KNOTS_P}.branding.patch"
+		eapply "${KNOTS_PATCHDIR}/${KNOTS_P}.ts.patch"
+		eapply "${FILESDIR}/${P}-fix_mempoolstats.patch"
+	fi
+
+	eapply_user
+
+	if ! use bitcoin_policy_rbf; then
+		sed -i 's/\(DEFAULT_ENABLE_REPLACEMENT = \)true/\1false/' src/validation.h || die
+	fi
+
+	echo '#!/bin/true' >share/genbuild.sh || die
+	mkdir -p src/obj || die
+	echo "#define BUILD_SUFFIX gentoo${PVR#${PV}}" >src/obj/build.h || die
+
+	sed -i 's/^\(Icon=\).*$/\1bitcoin-qt/;s/^\(Categories=.*\)$/\1P2P;Network;Qt;/' contrib/debian/bitcoin-qt.desktop || die
+
+	local filt= yeslang= nolang= lan ts x
+
+	for lan in $LANGS; do
+		lan="${lan/*:/}"
+		if [ ! -e src/qt/locale/bitcoin_$lan.ts ]; then
+			die "Language '$lan' no longer supported. Ebuild needs update."
+		fi
+	done
+
+	for ts in src/qt/locale/*.ts
+	do
+		x="${ts/*bitcoin_/}"
+		x="${x/.ts/}"
+		if ! use "$(bitcoin_lang2use "$x")"; then
+			nolang="$nolang $x"
+			rm "$ts" || die
+			filt="$filt\\|$x"
+		else
+			yeslang="$yeslang $x"
+		fi
+	done
+	filt="bitcoin_\\(${filt:2}\\)\\.\(qm\|ts\)"
+	sed "/${filt}/d" -i 'src/qt/bitcoin_locale.qrc' || die
+	sed "s/locale\/${filt}/bitcoin.qrc/" -i 'src/Makefile.qt.include' || die
+	einfo "Languages -- Enabled:$yeslang -- Disabled:$nolang"
+
+	eautoreconf
+	rm -r src/leveldb src/secp256k1 || die
+}
+
+src_configure() {
+	local my_econf=(
+		$(use_enable asm)
+		$(use_enable bip70)
+		$(use_with dbus qtdbus)
+		$(use_with libevent)
+		$(use_with qrcode qrencode)
+		$(use_with upnp miniupnpc)
+		$(use_enable upnp upnp-default)
+		$(use_enable test tests)
+		$(use_enable wallet)
+		$(use_enable zeromq zmq)
+		--with-gui=qt5
+		--disable-util-cli
+		--disable-util-tx
+		--disable-bench
+		--without-libs
+		--without-daemon
+		--disable-ccache
+		--disable-static
+		--with-system-leveldb
+		--with-system-libsecp256k1
+		--with-system-univalue
+	)
+	econf "${my_econf[@]}"
+}
+
+src_install() {
+	default
+
+	rm -f "${ED%/}/usr/bin/test_bitcoin" || die
+
+	insinto /usr/share/pixmaps
+	if use knots; then
+		newins "src/qt/res/rendered_icons/bitcoin.ico" "${PN}.ico"
+	else
+		newins "share/pixmaps/bitcoin.ico" "${PN}.ico"
+	fi
+	insinto /usr/share/applications
+	newins "contrib/debian/bitcoin-qt.desktop" "org.bitcoin.bitcoin-qt.desktop"
+
+	use libevent && dodoc doc/REST-interface.md doc/tor.md
+
+	use zeromq && dodoc doc/zmq.md
+
+	newbashcomp contrib/bitcoind.bash-completion ${PN}
+
+	if use kde; then
+		insinto /usr/share/kde4/services
```

Comment:
Given that KDE4 is on its way out, I think it would be more future-proof to install to 5, and symlink to 4. But that's your choice, I'm not going to insist.

Reply:
Installing this here makes it a no-op.

Reply:
Hmm, is there some kind of XDG standard for this yet?

Reply:
The place to be for Frameworks-based packages would be `/usr/share/kservices5`.

Reply:
bitcoin-qt is not a Frameworks-based package, just an ordinary Qt application with a URI handler.

Reply:
Let me put it like this: Nothing in the tree is going to look for content in `/usr/share/kde4/services`, there are only Frameworks-based packages looking for content in `/usr/share/kservices5`.